src/bidiMapper/domains/network/Cookie.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2024 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {Protocol} from 'devtools-protocol';
+
+import {
+  InvalidArgumentException,
+  UnsupportedOperationException,
+} from '../../../protocol/ErrorResponse';
+import {Network, type Storage} from '../../../protocol/protocol.js';
+
+const CDP_SPECIFIC_PREFIX = 'goog:';
+
+/**
+ * Utility class for presenting BiDi `network.Cookie` and it's conversion with
+ *  CDP's `Network.Cookie` and `Network.CookieParam`.
+ *  * https://w3c.github.io/webdriver-bidi/#type-network-Cookie
+ *  * https://chromedevtools.github.io/devtools-protocol/tot/Network/#type-Cookie
+ *  * https://chromedevtools.github.io/devtools-protocol/tot/Network/#type-CookieParam
+ */
+export class Cookie {
```

Comment:
Note: Usually only static Classes are forbidden/discouraged. as they are close to name spacing. Personally I have no problem, but we should align as a team on this and if we think it make sense disable relative rules for EsLint.

Reply:
https://google.github.io/styleguide/tsguide.html#container-classes

Reply:
Moved to `NetworkUtils`